pkg/event/internal/querytools/apply_query_field.go

```diff
@@ -13,13 +13,25 @@ import (
 )
 
 func ApplyQueryField(fieldPtr interface{}, queryField event.QueryField) error {
```

Comment:
I'm easily confused by the use of reflection, and I found it hard to follow this function. Can you add a docstring here to explain what happens and how?

Reply:
OK, will do.

Reply:
I've just added few comments. Feel free to express more concerns about readability/safety.